exasol/bucketfs/_buckets.py

```diff
@@ -223,6 +252,117 @@ def download(self, path: str, chunk_size: int = 8192) -> Iterable[ByteString]:
             yield from response.iter_content(chunk_size=chunk_size)
 
 
+class SaaSBucket:
+
+    def __init__(self, url: str, account_id: str, database_id: str, pat: str) -> None:
+        self._url = url
+        self._account_id = account_id
+        self.database_id = database_id
+        self._pat = pat
+
+    @property
+    def name(self) -> str:
+        return 'default'
+
+    @property
+    def udf_path(self) -> str:
+        return f'/buckets/uploads/{self.name}'
+
+    def files(self) -> Iterable[str]:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def delete(self, path: str) -> None:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def upload(self, path: str, data: ByteString | BinaryIO) -> None:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def download(self, path: str, chunk_size: int = 8192) -> Iterable[ByteString]:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def __str__(self):
+        return f"SaaSBucket<{self.name} | on: {self._url}>"
```

Comment:
maybe we should include the account_id and database_id, they are necessary to identify the bucket

Reply:
Both or just the database_id?